packages/vite/src/node/__tests__/utils.spec.ts

```diff
@@ -19,6 +20,32 @@ describe('injectQuery', () => {
     })
   }
 
+  test('relative path ""', () => {
+    expect(injectQuery('usr/vite/%20a%20', 'direct')).toEqual(
+      'usr/vite/%20a%20?direct'
+    )
+  })
+
+  test('relative path "./"', () => {
+    expect(injectQuery('./usr/vite/%20a%20', 'direct')).toEqual(
+      'usr/vite/%20a%20?direct'
+    )
+  })
+
+  test('file path', () => {
+    expect(injectQuery('file:///usr/vite/%20a%20', 'direct')).toMatch(
+      !isWindows
+        ? // d/a/vite/vite/file:/usr/vite/%20a%20?direct
+          new RegExp(
+            `${process.cwd().slice(1)}/file:/usr/vite/%20a%20\\?direct`
+          )
+        : // D:/a/vite/vite/file:/usr/vite/%20a%20?direct
```

Comment:
I expect these to be
```
file:///usr/vite/%20a%20?direct
file:///D:/usr/vite/%20a%20?direct

or

/usr/vite/%20a%20?direct
D:/usr/vite/%20a%20?direct
```

Maybe this bug has existed from before?

Reply:
yes, this PR not modify this scene and just add the test it. In fact, I also feel strange about this result 🤔